src/main.rs

```diff
@@ -77,43 +77,34 @@ struct Args {
     flag_version: bool,
 }
 
-fn write_dotfile_to_disk(filepath: &str, arena: treediff::Arena<String, String>) {
-    let mut dotfile = match File::create(&filepath) {
-        Ok(f) => f,
-        Err(_) => {
-            writeln!(&mut stderr(), "Could not create file {}.", &filepath).ok();
+fn write_dotfile_to_disk(filepath: &str, arena: &ast::Arena<String, String>) {
+    match emitters::write_dotfile_to_disk(filepath, arena) {
+        Ok(_) => (),
+        Err(emitters::EmitterError::CouldNotCreateFile) => {
+            writeln!(&mut stderr(), "Could not create file {}.", filepath).ok();
             process::exit(1);
         }
-    };
-    match arena.render_dotgraph(&mut dotfile) {
-        Ok(_) => (),
-        Err(_) => {
-            writeln!(&mut stderr(),
-                     "Could not write data to file file {}.",
-                     &filepath)
-                    .ok();
+        Err(emitters::EmitterError::CouldNotWriteToFile) => {
+            writeln!(&mut stderr(), "Could not write to file {}.", filepath).ok();
             process::exit(1);
         }
     };
 }
 
-fn parse_file(filename: &str,
-              lexer_path: &str,
-              yacc_path: &str)
-              -> treediff::ast::Arena<String, String> {
+fn parse_file(filename: &str, lexer_path: &str, yacc_path: &str) -> ast::Arena<String, String> {
```

Comment:
Does this really do the same thing as the original code? I think all the error messages that the user sees will be different?